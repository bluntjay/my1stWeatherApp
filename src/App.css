/*Add background-color, animation & a different font to the body*/
body {
  margin: 0;
  width: 100%;
  height: 100vh;
  font-family: "Exo", sans-serif;
  color: #fff;
  background: linear-gradient(-45deg, #1affff, #00cccc, #00bfff, #0086b3, #006080, #24478f,  #4775d1, #00001a);
  background-size: 400% 400%;
  animation: gradientBG 9s ease infinite;
}
/*Creates animation for the background-color*/
@keyframes gradientBG {
  0% {
    background-position: 0% 50%;
  }
  50% {
    background-position: 100% 50%;
  }
  100% {
    background-position: 0% 50%;
  }
}
/*Styles & positions main heading*/
#mainHead{
 text-shadow: 2px 2px 1px #5c5c8a, 5px 5px 1px rgba(0,0,0,0.15);
 margin-left: 30%;
 text-decoration: underline;
}
/*Styles & positions sub-heading*/
#subHead{
 margin-left: 34%;
 text-shadow: 1px 1px 0px #5c5c8a, 4px 4px 0px rgba(0,0,0,0.15);
 padding: 15px;

}
/*Styles & positions the div where my weather attributes are being written to*/
#foreCast{
  float: center;
  margin-left: 30%;
  text-align: center;
  background: linear-gradient( #4d004d, #862d86 60%);
  width: 30%;
  border-radius: 12px;
  box-shadow: 4px 4px 3px #000000, 8px 8px 4px rgba(20,15,10,5);

}
/*Styles each heading for the weather attributes*/
.weatherAspects{
  text-decoration: underline;
  text-shadow: 1px 1px 1px  #000000;
  padding: 9px;
}
/*Styles & positions the input for the search bars heading*/
#cityInp{
  text-shadow: 1px 1px 3px #00264d;
  margin-left: 13%;
  text-decoration: underline;
  padding: 3%;
  font-family: "Arial Black", Gadget, sans-serif; 
}
/*Styles & positions the search bar */
.searchBar{
  margin-left: 22%;
  width: 50%;
  border-radius: 12px 
}
/*Styles the search button*/
.searchBut{
  background-color: #e0e0eb;
  border-radius: 50%
}
#error{
  text-align: center;
  padding: 15px;
  font-size: 65px;
  text-shadow: 1px 1px 3px #00264d;
  text-decoration: underline;
}