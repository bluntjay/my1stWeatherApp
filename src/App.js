import React from 'react';
import './App.css';
// Import my different components from their files
import Tiles from "./tiles.js";
import Form from "./form.js";
import Weather from "./weather.js";
// Exports & returns class component to be rendered with it functionality 
export default class App extends React.Component {
  // Function that fetches API for weather as a JSON file, taking the city as an arguement
    getWeather = async (city) => {
        const APIcall = await fetch("http://api.openweathermap.org/data/2.5/weather?q=" + city + "&APPID=16a73ca4ce28ad2078b712e85c777b69")
        const response = await APIcall.json();
        console.log(response)

// Checks if there is an error message for not being able to find the city 
      if( response.cod === "404" ){
        // Sets a state for the error
        this.setState({
        error: "Couldn't find that city!",
        cod: "404"
        })
          return
        }
        // Sets each attribute of weather to the specified object key in JSON file 
        this.setState({
            temp: response.main.temp,
            max: response.main.temp_max,
            min: response.main.temp_min,
            city: response.name,
            lat: response.coord.lat,
            lon: response.coord.lon,
            windSpeed: response.wind.speed,
            windDir: response.wind.deg,
            country: response.sys.country,
            humidity: response.main.humidity,
            weather: response.weather[0].main,
            descrip: response.weather[0].description,
        });

    };

    render() {
        return (
            <div>
        <Tiles/>
        {/* Imports each of my components with their functionality */}
        <Form loadWeather={this.getWeather}/>
        {/* Error message will load if the cod = "404" - not being able find inputted city */}
        {this.state && this.state.cod === "404" ?
          <div id="error">{this.state.error}</div> :
          null}
          {/* The other aspects of the will load if theres no "404" erro */}
        {this.state && this.state.cod !== "404" ? (
        <Weather
          // Retrieves each weather attribute from objests in the JSON file
          temp={this.state.temp}
          max={this.state.max}
          min={this.state.min}
          city={this.state.city}
          country={this.state.country}
          lon={this.state.lon}
          lat={this.state.lat}
          windSpeed={this.state.windSpeed}
          windDir={this.state.windDir}
          humidity={this.state.humidity}
          weather={this.state.weather}
          descrip={this.state.descrip}
          error={this.state.error}
        />
        // If theres an error nothing will be rendered 
         ) : null}
      </div>


        )
    }
}