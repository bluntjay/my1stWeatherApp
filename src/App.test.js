import React from 'react';
import ReactDOM from 'react-dom';
import{App, APIcall, response, getWeather} from './App';

// Tests function that checkes if my fetch API function works properly
test.todo("APIcall"), done => {
	function callBack(data){
		expect(data).toBe("response").toMatchSnapshot();
		done();
}
	callBack()
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
	}



