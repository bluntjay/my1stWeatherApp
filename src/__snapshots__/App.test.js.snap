// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getWeather 1`] = `[Function]`;
